src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityReservoirHatch.java

```diff
@@ -46,17 +50,24 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.util.Arrays;
 import java.util.List;
 
 public class MetaTileEntityReservoirHatch extends MetaTileEntityMultiblockNotifiablePart
-                                          implements IMultiblockAbilityPart<IFluidTank> {
+                                          implements IMultiblockAbilityPart<IFluidTank>,
+                                          IGhostSlotConfigurable {
 
     private static final int FLUID_AMOUNT = 2_000_000_000;
     private final InfiniteWaterTank fluidTank;
+    private GhostCircuitItemStackHandler circuitInventory;
 
     public MetaTileEntityReservoirHatch(ResourceLocation metaTileEntityId) {
         super(metaTileEntityId, GTValues.EV, false);
         this.fluidTank = new InfiniteWaterTank(getInventorySize(), this);
+        if (this.hasGhostCircuitInventory()) {
+            this.circuitInventory = new GhostCircuitItemStackHandler(this);
+            this.circuitInventory.addNotifiableMetaTileEntity(this);
```

Comment:
Same question about notifiable here.

Also, does this now need to override `addToMultiblock` and remove, to deal with the ghost slot, like the fluid hatch?